### PR TITLE
Correct misleading wording in Pod DNS Policies

### PR DESCRIPTION
**Pull request by chrisseto**
A debugging rabbit hole lead me to digging around kublet's behavior regarding certain DNS Policies.

This commit updates some potentially misleading wordings to more correctly convey kubelet's behavior which was discovered after some rabbit hole debugging of DNS Policies.

The ClusterFirst policy documentation would imply that the resolve.conf of a Pod would contain a fallback to the Node's nameserver. This is not the cause, [kubelet explicitly mentions](https://github.com/kubernetes/kubernetes/blob/4276ed36282405d026d8072e0ebed4f1da49070d/pkg/kubelet/network/dns/dns.go#L403-L407) that the fallback is handled by whatever system is running kube-dns or coredns which dramatically changes the behavior of the system when the cluster's DNS is unavailable.

The ClusterFirstWithHostNet policy is lacking any amount of detail as to why one would desire to use it. This commit doesn't go into complete detail but adds an important note about the [behavior of `hostNetwork` when not using the `ClusterFirstWIthHostNet` Policy](https://github.com/kubernetes/kubernetes/blob/4276ed36282405d026d8072e0ebed4f1da49070d/pkg/kubelet/network/dns/dns.go#L311-L318).

### CODE REVIEW

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 1]
```diff
@@ -221,12 +221,12 @@ following Pod-specific DNS policies. These policies are specified in the
   for more details.
 - "`ClusterFirst`": Any DNS query that does not match the configured cluster
   domain suffix, such as "`www.kubernetes.io`", is forwarded to the upstream
-  nameserver inherited from the node. Cluster administrators may have extra
+  nameserver inherited from the DNS Service's node. Cluster administrators may have extra
```
Comment (thockin) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:224]: I would change this even more, like:

"""
...is forwarded to an upstream nameserver by the DNS server.
"""

File: content/en/docs/concepts/services-networking/dns-pod-service.md [hunk 2]
```diff
@@ -221,12 +221,12 @@ following Pod-specific DNS policies. These policies are specified in the
...
   stub-domain and upstream DNS servers configured.
   See [related discussion](/docs/tasks/administer-cluster/dns-custom-nameservers)
   for details on how DNS queries are handled in those cases.
 - "`ClusterFirstWithHostNet`": For Pods running with hostNetwork, you should
-  explicitly set its DNS policy "`ClusterFirstWithHostNet`".
+  explicitly set its DNS policy to "`ClusterFirstWithHostNet`". Otherwise, Pods running with hostNetwork and `"ClusterFirst"` will fallback to the behavior of the `"Default"` policy.
```
Comment (thockin) [inline content/en/docs/concepts/services-networking/dns-pod-service.md:229]: linewrap ?

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 1 above)
Comment (chrisseto) [reply content/en/docs/concepts/services-networking/dns-pod-service.md]: Done! Thanks for the improved wording.

File: content/en/docs/concepts/services-networking/dns-pod-service.md (hunk 2 above)
Comment (chrisseto) [reply content/en/docs/concepts/services-networking/dns-pod-service.md]: Fixed and fixed line wrapping in the commit message as well.

**Review by chrisseto — COMMENTED** [review]
TFTR!

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Review by Arhell — COMMENTED** [review]
/lgtm

**Pull request merged**